File: stack_using_linkedlist.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

struct Node
{
    int data;
    struct Node *next;
};

struct Node *head;

void push(int data)
{
    struct Node *temp = (struct Node *)malloc(sizeof(struct Node));
    temp->data = data;
    temp->next = head;
    head = temp;
}

bool isEmpty()
{
    if (head == NULL)
        return true;
    return false;
}

int pop()
{
    if (!isEmpty())
    {
        int data = head->data;
        struct Node *temp = head;
        head = head->next;
        free(temp);
        return data;
    }
}

int top()
{
    if (!isEmpty())
        return head->data;
}

void print()
{
    struct Node *temp = head;
    while (temp != NULL)
    {
        printf("%d ", temp->data);
        temp = temp->next;
    }
}

// int main()
// {
//     push(3);
//     push(2);
//     push(1);
//     printf("%d ", pop());
//     printf("%d ", pop());
//     printf("%d ", pop());
//     printf("\n");
//     print();
// }

File: stack_using_linkedlist.js
let Node = function(data, next = null){
    this.data = data;
    this.next = next;
}

let MinStack = function(){
    this.head = null;
}

MinStack.prototype.push = (data) => {
    let node = new Node(data);
    if (!this.head) {
        this.head = node;
        return;
    }
    node.next = this.head;
    this.head = node;
}

MinStack.prototype.pop = () => {
    if (!this.head)
        return;
    this.head = this.head.next;
}

MinStack.prototype.top = () => {
    if (!this.head)
        return null;
    return this.head.data;
}

MinStack.prototype.getMin = () => {
    if (!this.head)
        return null;
    let current = this.head;
    let min = Number.POSITIVE_INFINITY;
    while (current != null) {
        if (current.data < min)
            min = current.data;
        current = current.next;
    }
    return min;
}

let stack = new MinStack();
stack.push(-200);
stack.push(1);
stack.push(2);
console.log(stack.getMin());
console.log(stack.top());
stack.pop();
console.log(stack.top());
